### ISSUE TITLE
Get xid of window

### ISSUE THREAD

**Issue by johnlane** [ruby-gnome2/ruby-gnome2#999]
The Python bindings for Gtk allow a window's xid (X11 window id) to be obtained:

    self.get_toplevel().get_window().get_xid()

But this is not possible in Ruby

    toplevel.window.xid

gives

    undefined method `xid' for #<#<Class:0x000000038e40b0>:0x3975b00 ptr=0x28a64d0> (NoMethodError)

As [raised on ML](https://sourceforge.net/p/ruby-gnome2/mailman/ruby-gnome2-devel-en/thread/1ce7b201-a0fb-a355-fe6b-3992621d1ff2%40lane.uk.net):

> It seems that it is not yet supported in the Gdk gem. Maybe we will just have to create a GObjectIntrospectionloader for GdkX11.

**Comment by kou** [comment]
I've add X11 related method support.

Here is a workaround until the next release:

```ruby
require "gtk3"

Gtk3.init

require "gi"
GI::Loader.load("GdkX11", Gdk)

# Now you can use Gdk::X11Window#xid.
```

**Comment by johnlane** [comment]
Just confirming that works, but I didn't need `Gtk3.init`. In fact, including it resulted in an error: `in '<main>': uninitialized constant Gtk3 (NameError)`

**Comment by kou** [comment]
Ah, sorry. It's a typo. It should be `Gtk.init`.

**Comment by johnlane** [comment]
This is just a comment. In the above `require gi` got me confused because I had a completely unrelated gem called [gi](https://rubygems.org/gems/gi/versions/0.1.0) installed. I found that I did not need `Gtk.init` - my program worked without it... until I removed that unrelated gem. After that, I needed `Gtk.init`for the above to work. Even though there is no mention of `GI` (uppercase) in that gem. Not an issue - I was just conused for a moment. Your workaround works as directed.
